mos_tests/nfv/conftest.py

```diff
@@ -0,0 +1,120 @@
+#    Copyright 2016 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import pytest
+
+from mos_tests.functions import common
+
+
+@pytest.yield_fixture
+def aggregate(os_conn):
+    hp_computes = []
+    for compute in os_conn.env.get_nodes_by_role('compute'):
+        with os_conn.env.get_ssh_to_node(compute.data['ip']) as remote:
+            res = remote.execute(
+                'grep HugePages_Total /proc/meminfo')['stdout']
+        if res:
+            if res[0].split(':')[1].rstrip().lstrip() != '0':
+                hp_computes.append(compute)
+    aggr = os_conn.nova.aggregates.create('hpgs-aggr', 'nova')
+    os_conn.nova.aggregates.set_metadata(aggr, {'hpgs': 'true'})
+    for host in hp_computes:
+        os_conn.nova.aggregates.add_host(aggr, host.data['fqdn'])
+    yield aggr
+    for host in hp_computes:
+        os_conn.nova.aggregates.remove_host(aggr, host.data['fqdn'])
+    os_conn.nova.aggregates.delete(aggr)
+
+
+@pytest.yield_fixture()
+def nfv_flavor(os_conn, clear_old, request):
+    flavors = getattr(
+        request, 'param', [[['m1.small.hpgs', 512, 1, 1],
+                           [{'hw:mem_page_size': 2048}, ]], ])
+    created_flavors = []
+
+    for flavor_params, flavor_keys in flavors:
+        flavor = os_conn.nova.flavors.create(*flavor_params)
+        for flavor_key in flavor_keys:
+            flavor.set_keys(flavor_key)
+        created_flavors.append(flavor)
+    yield created_flavors
+    for flavor in created_flavors:
+        os_conn.nova.flavors.delete(flavor.id)
+
+
+@pytest.yield_fixture(scope="class")
+def keypair(os_conn):
+    key = os_conn.create_key(key_name='nfv_key')
+    yield key.name
```

Comment:
AFAIK you can't retrieve private key from nova using only keypair name. You shold preserve all created keypair object (otherwise you won't connect to vm with this key).